src/containers/App/SubjectmatterPage.jsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2016-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree. *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Route, Switch, withRouter } from 'react-router-dom';
+import { injectT } from 'ndla-i18n';
+import { TopicArticle, LearningResource, Concept } from 'ndla-icons/editor';
+import NotFoundPage from '../NotFoundPage/NotFoundPage';
+import PrivateRoute from '../PrivateRoute/PrivateRoute';
+import TopicArticlePage from '../TopicArticlePage/TopicArticlePage';
+import LearningResourcePage from '../LearningResourcePage/LearningResourcePage';
+import TypeMenu from './TypeMenu';
+
+const DEFAULT_TYPE = 'learning-resource';
+
+class SubjectmatterPage extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      type: DEFAULT_TYPE,
+    };
+  }
+  componentWillMount() {
+    const { location: { pathname } } = this.props;
+    const splittedPathname = pathname.split('/');
+    const type =
+      splittedPathname.length > 3 ? splittedPathname[2] : DEFAULT_TYPE;
+    this.setState({ type });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { location: { pathname } } = nextProps;
```

Comment:
Det her så litt hårete ut (og er duplisert to steder). Du kan ikke bruke https://reacttraining.com/react-router/web/api/NavLink ?